<launch>
	<node name="dolly_base_node" pkg="dolly_base" type="dolly_base_node" output="screen" />

	<node name="stm_topic_converter" pkg="dolly_base" type="stm_topic_converter_node" output="screen" />

	<!--<node name="hardware_emulator_node" pkg="dolly_base" type="hardware_emulator_node" output="screen" /> -->

	<node name="pid_controller" pkg="pid" ns="/dolly/left_wheel" type="controller" >
		<param name="Kp" value="0.25" />
		<param name="Ki" value="5.0" />
		<param name="Kd" value="0.001" />
		<param name="upper_limit" value="50" />
		<param name="lower_limit" value="-50" />
		<param name="windup_limit" value="10" />
		<param name="max_loop_frequency" value="105.0" />
		<param name="min_loop_frequency" value="20.0" />
		<remap from="setpoint" to="/dolly/left_wheel/cmd_vel64" />
		<remap from="state" to="/dolly/left_wheel/cur_vel64" />
		<remap from="control_effort" to="/dolly/left_wheel/pwd64" />
    </node>

	<node name="pid_controller" pkg="pid" ns="/dolly/right_wheel" type="controller" >
		<param name="Kp" value="0.25" />
		<param name="Ki" value="5.0" />
		<param name="Kd" value="0.001" />
		<param name="upper_limit" value="50" />
		<param name="lower_limit" value="-50" />
		<param name="windup_limit" value="10" />
		<param name="max_loop_frequency" value="105.0" />
		<param name="min_loop_frequency" value="20.0" />
		<remap from="setpoint" to="/dolly/right_wheel/cmd_vel64" />
		<remap from="state" to="/dolly/right_wheel/cur_vel64" />
		<remap from="control_effort" to="/dolly/right_wheel/pwd64" />
    </node>		
</launch>
